### ISSUE TITLE
"oc export -o wide …" panics

### ISSUE THREAD

**Issue by hansmi** [openshift/origin#17163]
Running `oc export -o wide` with any resource type panics in `pkg/cmd/cli/cmd/export.go`.

##### Version
```
oc v3.6.173.0.49
kubernetes v1.6.1+5115d708d7
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://…:8443
openshift v3.6.173.0.49
kubernetes v1.6.1+5115d708d7
```

Also reproduced on Origin 1.5.1 and OpenShift Container Platform 3.6.173.0.5.

##### Steps To Reproduce
1. `oc export -o wide pod`

##### Current Result
```
# oc export pod -o wide
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x5ace72]

goroutine 1 [running]:
panic(0x37ffd80, 0xc420014100)
        /usr/lib/golang/src/runtime/panic.go:500 +0x1a1
github.com/openshift/origin/pkg/cmd/cli/cmd.RunExport(0xc4200136b0, 0x6354360, 0x6501c30, 0x634ba20, 0xc420084000, 0x634ba60, 0xc420084008, 0xc420419440, 0xc420b9acc0, 0x1, ...)
        /builddir/build/BUILD/atomic-openshift-git-0.0b9377a/_output/local/go/src/github.com/openshift/origin/pkg/cmd/cli/cmd/export.go:186 +0xfc2
github.com/openshift/origin/pkg/cmd/cli/cmd.NewCmdExport.func1(0xc420419440, 0xc420b9acc0, 0x1, 0x3)
        /builddir/build/BUILD/atomic-openshift-git-0.0b9377a/_output/local/go/src/github.com/openshift/origin/pkg/cmd/cli/cmd/export.go:63 +0xc7
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).execute(0xc420419440, 0xc420b9ac30, 0x3, 0x3, 0xc420419440, 0xc420b9ac30)
        /builddir/build/BUILD/atomic-openshift-git-0.0b9377a/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:603 +0x439
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4202b26c0, 0xc420084000, 0xc420084010, 0xc4202b26c0)
        /builddir/build/BUILD/atomic-openshift-git-0.0b9377a/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:689 +0x367
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4202b26c0, 0x2, 0xc4202b26c0)
        /builddir/build/BUILD/atomic-openshift-git-0.0b9377a/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.main()
        /builddir/build/BUILD/atomic-openshift-git-0.0b9377a/_output/local/go/src/github.com/openshift/origin/cmd/oc/oc.go:36 +0x196
```

##### Expected Result
A human-readable error message, i.e. `error: output format "wide" not supported`.

**Comment by xiaods** [comment]
upstream is works.

```
$ oc get pod
NAME               READY     STATUS      RESTARTS   AGE
httpd-ex-1-build   0/1       Completed   0          2m
httpd-ex-1-qx8db   1/1       Running     0          26s
xiaods at XiaoTommydeMacBook-Pro in ~/go/src/github.com/openshift/origin on devel*
$ oc export -o wide pod
NAME               AGE
httpd-ex-1-build   <unknown>
                   <unknown>
```

**Comment by smarterclayton** [comment]
@openshift/sig-user-interface

**Comment by juanvallejo** [comment]
@fabianofranz based on https://github.com/openshift/origin/issues/17163#issuecomment-341625454 and local testing, the panic only happens in 3.6 clients.
Will open a patch against that branch.

**Comment by hansmi** [comment]
@juanvallejo It happens in 3.5 clients too, unfortunately:

```
# oc version
oc v3.5.5.31
kubernetes v1.5.2+43a9be4
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://master-int:8443
openshift v3.5.5.31
kubernetes v1.5.2+43a9be4
```

```
# oc export -o wide pod
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x5933a2]

goroutine 1 [running]:
panic(0x1f98720, 0xc4200120d0)
	/usr/lib/golang/src/runtime/panic.go:500 +0x1a1
github.com/openshift/origin/pkg/cmd/cli/cmd.RunExport(0xc4203fe2a0, 0x3602340, 0x3759b40, 0x35fcd40, 0xc420028008, 0x35fcd80, 0xc420028010, 0xc42023cd80, 0xc420b84fc0, 0x1, ...)
	/builddir/build/BUILD/atomic-openshift-git-0.b6f55a2/_output/local/go/src/github.com/openshift/origin/pkg/cmd/cli/cmd/export.go:176 +0xe82
github.com/openshift/origin/pkg/cmd/cli/cmd.NewCmdExport.func1(0xc42023cd80, 0xc420b84fc0, 0x1, 0x3)
	/builddir/build/BUILD/atomic-openshift-git-0.b6f55a2/_output/local/go/src/github.com/openshift/origin/pkg/cmd/cli/cmd/export.go:61 +0xc7
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).execute(0xc42023cd80, 0xc420b84f30, 0x3, 0x3, 0xc42023cd80, 0xc420b84f30)
	/builddir/build/BUILD/atomic-openshift-git-0.b6f55a2/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:603 +0x439
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420362480, 0xc420028008, 0xc420028018, 0xc420362480)
	/builddir/build/BUILD/atomic-openshift-git-0.b6f55a2/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:689 +0x367
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420362480, 0x2, 0xc420362480)
	/builddir/build/BUILD/atomic-openshift-git-0.b6f55a2/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.main()
	/builddir/build/BUILD/atomic-openshift-git-0.b6f55a2/_output/local/go/src/github.com/openshift/origin/cmd/oc/oc.go:36 +0x196
```

**Comment by juanvallejo** [comment]
Lowering severity as this does not affect >= 3.7 client versions

**Comment by hansmi** [comment]
/remove-lifecycle stale

@juanvallejo, it appears as if your changes weren't merged. An `oc` binary versioned `v3.10.0-alpha.0+30a3676-736` still fails, though not with a panic. I guess this isn't supposed to happen and `-o ''` (the default) works as expected. Should I consider this to "work as designed"?

```
$ oc version
oc v3.10.0-alpha.0+30a3676-736
kubernetes v1.10.0+b81c8f8
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://172.30.0.1:443
openshift v3.6.173.0.96
kubernetes v1.6.1+5115d708d7

$ oc export pod -o wide
error: unknown type &v1.List{TypeMeta:v1.TypeMeta{Kind:"List", APIVersion:"v1"}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:""}, Items:[]runtime.RawExtension{runtime.RawExtension{Raw:[]uint8(nil), Object:(*v1.Pod)(0xc42100ca80)}, […]}}}
```

**Comment by juanvallejo** [comment]
@hansmi changes now merged into 3.6. Currently no plan to support `wide` printer for this command. The error message you see will improve in the next rebase as part of ongoing upstream work on the printers.
Thanks

**Comment by hansmi** [comment]
@juanvallejo Thank you for the feedback. Considering this closed then as the command is no longer panicking.
